include/pybind11/stl_bind.h

```diff
@@ -694,6 +694,26 @@ struct ItemsViewImpl : public detail::items_view {
     Map &map;
 };
 
+template <typename KeyType>
+std::string format_message_key_error(const KeyType &k) {
+    std::string message;
+    try {
+        message = str(cast(k));
+    } catch (const std::exception &) {
+        try {
+            message = repr(cast(k));
+        } catch (const std::exception &) {
+            return message;
+        }
+    }
+    const size_t max_length = 80;
+    if (message.length() > max_length) {
+        message.resize(max_length);
+        return message + "...";
```

Comment:
OK as-is if you don't want to spend the time, but it would be nice to go the extra mile (with the idea again that this may well be reused).

Main idea: Very often the interesting part is either at the beginning or the end, rarely in the middle.

I'd be a little more generous, I think Python uses a `%.200s` a lot.

Goal (roughly): `"first 100 chars ... last 100 chars"`

Reply:
Done, PTAL.